### ISSUE TITLE
cannot load such file -- jekyll/post (LoadError)

### ISSUE THREAD

**Issue by csillag** [jekyll/jekyll-redirect-from#77]
I am getting this, while adding this gem to a jekyll (3.0.0) project:

```
/whatever/bundle/ruby/2.1.0/gems/jekyll-redirect-from-0.8.0/lib/jekyll-redirect-from/redirector.rb:41:in `is_dynamic_document?': cannot load such file -- jekyll/post (LoadError)
	from /whatever/bundle/ruby/2.1.0/gems/jekyll-redirect-from-0.8.0/lib/jekyll-redirect-from/redirector.rb:47:in `has_alt_urls?'
	from /whatever/bundle/ruby/2.1.0/gems/jekyll-redirect-from-0.8.0/lib/jekyll-redirect-from/redirector.rb:14:in `block in generate_alt_urls'
	from /whatever/bundle/ruby/2.1.0/gems/jekyll-3.0.0/lib/jekyll/collection.rb:37:in `each'
	from /whatever/bundle/ruby/2.1.0/gems/jekyll-3.0.0/lib/jekyll/collection.rb:37:in `public_send'
	from /whatever/bundle/ruby/2.1.0/gems/jekyll-3.0.0/lib/jekyll/collection.rb:37:in `method_missing'
	from /whatever/bundle/ruby/2.1.0/gems/jekyll-redirect-from-0.8.0/lib/jekyll-redirect-from/redirector.rb:13:in `generate_alt_urls'
	from /whatever/bundle/ruby/2.1.0/gems/jekyll-redirect-from-0.8.0/lib/jekyll-redirect-from/redirector.rb:7:in `generate'
	from /whatever/bundle/ruby/2.1.0/gems/jekyll-3.0.0/lib/jekyll/site.rb:154:in `block in generate'
	from /whatever/bundle/ruby/2.1.0/gems/jekyll-3.0.0/lib/jekyll/site.rb:153:in `each'
	from /whatever/bundle/ruby/2.1.0/gems/jekyll-3.0.0/lib/jekyll/site.rb:153:in `generate'
	from /whatever/bundle/ruby/2.1.0/gems/jekyll-3.0.0/lib/jekyll/site.rb:58:in `process'
	from /whatever/bundle/ruby/2.1.0/gems/jekyll-3.0.0/lib/jekyll/command.rb:28:in `process_site'
	from /whatever/bundle/ruby/2.1.0/gems/jekyll-3.0.0/lib/jekyll/commands/build.rb:60:in `build'
	from /whatever/bundle/ruby/2.1.0/gems/jekyll-3.0.0/lib/jekyll/commands/build.rb:35:in `process'
	from /whatever/bundle/ruby/2.1.0/gems/jekyll-3.0.0/lib/jekyll/commands/build.rb:18:in `block (2 levels) in init_with_program'
	from /whatever/bundle/ruby/2.1.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `call'
	from /whatever/bundle/ruby/2.1.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `block in execute'
	from /whatever/bundle/ruby/2.1.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `each'
	from /whatever/bundle/ruby/2.1.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `execute'
	from /whatever/bundle/ruby/2.1.0/gems/mercenary-0.3.5/lib/mercenary/program.rb:42:in `go'
	from /whatever/bundle/ruby/2.1.0/gems/mercenary-0.3.5/lib/mercenary.rb:19:in `program'
	from /whatever/bundle/ruby/2.1.0/gems/jekyll-3.0.0/bin/jekyll:17:in `<top (required)>'
	from /home/csillag/kapisztran/orasite/admin/vendor/bundle/bin/jekyll:16:in `load'
	from /home/csillag/kapisztran/orasite/admin/vendor/bundle/bin/jekyll:16:in `<main>'
```

I haven't defined any redirects yet; I have only added the gem to `_config.yml`.
What am I missing?

Thanks.

**Comment by parkr** [comment]
Fixed by #76.

**Comment by bradonomics** [comment]
While this is a closed issue, I was getting the same error. I recently upgraded Ruby and Jekyll so I'm not sure about the fix in #76.

I'm running Linux Mint 17.2. My ruby and jekyll versions are:
```
$ ruby --version
ruby 2.2.3p173 (2015-08-18 revision 51636) [x86_64-linux-gnu]
$ jekyll --version
jekyll 3.0.1
```
When I first tried to do the build I was getting the following:
```
$ jekyll build --watch
Configuration file: /home/brad/GitHub/dev.project.github.io/_config.yml
/usr/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require': cannot load such file -- jekyll-redirect-from (LoadError)
	from /usr/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /var/lib/gems/2.2.0/gems/jekyll-3.0.1/lib/jekyll/plugin_manager.rb:30:in `block in require_gems'
	from /var/lib/gems/2.2.0/gems/jekyll-3.0.1/lib/jekyll/plugin_manager.rb:27:in `each'
	from /var/lib/gems/2.2.0/gems/jekyll-3.0.1/lib/jekyll/plugin_manager.rb:27:in `require_gems'
	from /var/lib/gems/2.2.0/gems/jekyll-3.0.1/lib/jekyll/plugin_manager.rb:19:in `conscientious_require'
	from /var/lib/gems/2.2.0/gems/jekyll-3.0.1/lib/jekyll/site.rb:97:in `setup'
	from /var/lib/gems/2.2.0/gems/jekyll-3.0.1/lib/jekyll/site.rb:49:in `initialize'
	from /var/lib/gems/2.2.0/gems/jekyll-3.0.1/lib/jekyll/commands/build.rb:30:in `new'
	from /var/lib/gems/2.2.0/gems/jekyll-3.0.1/lib/jekyll/commands/build.rb:30:in `process'
	from /var/lib/gems/2.2.0/gems/jekyll-3.0.1/lib/jekyll/commands/build.rb:18:in `block (2 levels) in init_with_program'
	from /var/lib/gems/2.2.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `call'
	from /var/lib/gems/2.2.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `block in execute'
	from /var/lib/gems/2.2.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `each'
	from /var/lib/gems/2.2.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `execute'
	from /var/lib/gems/2.2.0/gems/mercenary-0.3.5/lib/mercenary/program.rb:42:in `go'
	from /var/lib/gems/2.2.0/gems/mercenary-0.3.5/lib/mercenary.rb:19:in `program'
	from /var/lib/gems/2.2.0/gems/jekyll-3.0.1/bin/jekyll:17:in `<top (required)>'
	from /usr/local/bin/jekyll:23:in `load'
	from /usr/local/bin/jekyll:23:in `<main>'
```
I them tried installing bundle per [the conversation on a different thread](https://github.com/twbs/bootstrap/issues/17254), but that had no effect.

I then tried manually installing jekyll-redirect-from (`gem install jekyll-redirect-from`) then trying the build again and finally it worked.

Since jekyll-redirect-from is available in GitHub Pages, I would think adding it in the default Jekyll install would be best and only having to manually install gems/plugins not allowed on GitHub Pages.

**Comment by parkr** [comment]
> Since jekyll-redirect-from is available in GitHub Pages, I would think adding it in the default Jekyll install would be best and only having to manually install gems/plugins not allowed on GitHub Pages.

Sounds like what you're looking for is the `github-pages` gem. [It's built for that purpose](https://github.com/github/pages-gem#project-goals).
